### PR TITLE
TechDocs: Use interface and not implementation in ApiRef<T>

### LINKED ISSUE #4404: TechDocs ApiRef<T> should use interface and not implementation
The current `techdocsStorageApiRef` & `techdocsApiRef` use class implementations named `TechDocsStorageApi` & `TechDocsApi` which are not interfaces. This seems incorrect and confusing since the class implementation is suffixed with `Api` which is not common throughout the code base. Instead the interface is more typically used as the ApiRef type and is suffixed with `Api`

## Expected Behavior

The `techdocsStorageApiRef` & `techdocsApiRef` objects use interfaces instead of the concrete class implementations.

## Current Behavior

The `techdocsStorageApiRef` & `techdocsApiRef` objects use class implementations.

## Possible Solution

- Rename class `TechDocsStorageApi` to `TechDocsStorageApiImpl`.
- Rename class `TechDocsApi` to `TechDocsApiImpl`.
- Rename interface `TechDocsStorage` to `TechDocsStorageApi`.
- Rename interface `TechDocs` to `TechDocsApi`.
- Update relevant changes accordingly

I took a quick stab at this, but there are some problems introduced by this assumption within the current code base which needs untangling.

### PR DESCRIPTION
**Pull request by Fox32**
Closes #4404

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: app-config.yaml [hunk 1]
```diff
@@ -2,9 +2,9 @@ app:
   title: Backstage Example App
   baseUrl: http://localhost:3000
   googleAnalyticsTrackingId: # UA-000000-0
-  datadogRum:
-    clientToken: '123456789'
-    applicationId: qwerty
```
Comment (Fox32) [inline app-config.yaml:7]: See https://discord.com/channels/687207715902193673/687235481154617364/834739375471329300

**Review by iamEAP — APPROVED** [review]
Nice!  Makes sense and all looks good to me.  You're really killing it this week, @Fox32!  Thanks.

**Comment by Fox32** [comment]
I resolved the conflicts

**Review by soapraj — APPROVED** [review]

File: .changeset/techdocs-swift-waves-sleep.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+---
+'@backstage/plugin-techdocs': patch
```
Comment (Rugvip) [inline .changeset/techdocs-swift-waves-sleep.md:2]: The existing types are exported, so expecting that this would need to be `minor`

**Review by Rugvip — COMMENTED** [review]
Thanks, LGTM! 👍 

Just a changeset thang :p

**Review by Rugvip — APPROVED** [review]
:shipit:

**Pull request merged**